### DISCUSSION TITLE
Idea for course in Old English

### DISCUSSION THREAD

**Discussion by nomad8809** [kantord/LibreLingo#3219] [Q&A]
Hello,

I hope all is well! I wanted to quickly introduce myself and my interest.

Next year, I will be taking coursework in Old English. I'm trying to get ahead and start studying now, but I'm having an issue. I really want 'DuoLingo for Old English' to get in frequent repetition, but they don't offer it. I have used DuoLingo for French and German and absolutely love it. Flash cards (Anki) works fine for individual words, but there is a large number of variations due to inflections, variations, etc. I want to be able to practice putting sentences together the way it's done through Duolingo.

This project looks fantastic. I'm an Electrical Engineer, not a software developer, so I don't have the skills to create an app. However, looking through the documentation, I believe I'm capable of putting a course together in parallel with my learning. I'd also need to get an instructor or two on-board to review all the content for accuracy. If this all sounds okay with you, I just have a couple of questions regarding the easiest methods of creating contents. Looking through the examples, it looks like I'm limited to direct entries on phrases. I want to make sure I understand the structure of courses clearly. I also have questions on font and audio.

Thank you!

Charles Turner

**Comment by kantord** [comment]
> Hello,
> 
> I hope all is well! I wanted to quickly introduce myself and my interest.
> 
> Next year, I will be taking coursework in Old English. I'm trying to get ahead and start studying now, but I'm having an issue. I really want 'DuoLingo for Old English' to get in frequent repetition, but they don't offer it. I have used DuoLingo for French and German and absolutely love it. Flash cards (Anki) works fine for individual words, but there is a large number of variations due to inflections, variations, etc. I want to be able to practice putting sentences together the way it's done through Duolingo.
> 
> This project looks fantastic. I'm an Electrical Engineer, not a software developer, so I don't have the skills to create an app. However, looking through the documentation, I believe I'm capable of putting a course together in parallel with my learning. I'd also need to get an instructor or two on-board to review all the content for accuracy. If this all sounds okay with you, I just have a couple of questions regarding the easiest methods of creating contents. Looking through the examples, it looks like I'm limited to direct entries on phrases. I want to make sure I understand the structure of courses clearly. I also have questions on font and audio.
> 
> Thank you!
> 
> Charles Turner

Hi, so what are your concrete questions?

> Looking through the examples, it looks like I'm limited to direct entries on phrases.

What do you mean about "direct entries"?

**Comment by nomad8809** [comment]
To get started:

Are the phrases I provide in the file the only phrases that are tested, or is there more to it? I think with Duolingo, they have standard phrases for each module, but then they are modified as they are presented to the user. It looks like throughout these challenges, parts of the phrase such as names or personal pronouns and the corresponding verbs are modified. Or if the name is masculine or feminine, a corresponding adjective may change. Is there something like this in LibreLingo, or would I have to provide every phrase variant manually?

Also, can I provide an IPA spelling of words so the TTS will work correctly?

Thank you!

**Reply by kantord** [reply]
> they have standard phrases for each module, but then they are modified as they are presented to the user. It looks like throughout these challenges, parts of the phrase such as names or personal pronouns and the corresponding verbs are modified. 

so you mean basically a template for sentences? I think that we have something like this, but I was looking for the documentation and could not find it myself :joy_cat:  have to look in the code.

> Also, can I provide an IPA spelling of words so the TTS will work correctly?

No, this is not a feature currently.

**Reply by kantord** [reply]
ok, regarding the templating system (variants of sentences) I found that this is not implemented yet, but there has been deep research into how to implement it. It just so happened that nobody actually made a pull request yet.

Since you are interested in this feature, here is the issue link if you want to add your 2 cents: https://github.com/LibreLingo/LibreLingo/issues/1122 since it seems like at the moment there is not working implementation submitted yet, it's still time to provide your input and influence the final outcome if needed
